### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by PromzzyKoncepts**
# Hey There :
> In this PR, we successfully:

- [x] Refactored the code using DRY, YAGNI and KISS
- [x] Ensured the code corresponds to SOLID principles

### Thanks

### CODE REVIEW

**Review by ghost — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights

✅ Correct use of Gitflow
✅ No linter error


## Required Changes ♻️

- [ ] Since you have not deployed the project, Kindly remove the `live demo link` from your readme file to keep your readme file professional
- [ ] Kindly provide the correct instructions to run the project locally as you included  `run  rails server` yet this is not a rails project
- [ ] Kindly note that the `FAQ` section is optional. You can fill it or not include it in your readme file

## Optional suggestions

N/A
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 1]
```diff
@@ -98,16 +86,7 @@ Install this project with:
...
   gem install
 ```
--
 
-### Usage
-
-To run the project, execute the following command:
-
-```sh
-  rails server
```
Comment (tugsem) [inline README.md:108]: - [OPTIONAL]  Kindly include the usage section and replace `rails server` with `ruby main.rb`

File: README.md [hunk 2]
```diff
@@ -98,16 +86,7 @@ Install this project with:
   cd my-project
```
Comment (tugsem) [inline README.md:86]: -  [OPTIONAL] Kindly replace `my-project` with `school-library`

**Review by tugsem — APPROVED** [review]
## STATUS: APPROVED 👍🎯  
Hi @PromzzyKoncepts , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⭐ 
✔️ You have implemented requested changes
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**